OSDOCS#17487: Trust Manager operand for cert-manager (#108188)
openshift/openshift-docs: security/cert_manager_operator/cert-manager-trust-manager.adoc

:_mod-docs-content-type: ASSEMBLY
[id="cert-manager-trust-manager"]
= Distributing certificates by using trust-manager operand
include::_attributes/common-attributes.adoc[]
:context: cert-manager-trust-manager

[role="_abstract"]
The trust-manager operand simplifies the distribution of certificate authority (CA) certificates across {product-title} clusters. As an administrator, you can configure the operand according to the cluster requirements and manage trust bundles efficiently.

:FeatureName: Distributing certificates by using trust manager
include::snippets/technology-preview.adoc[leveloffset=+1]

The trust-manager operand provides the following benefits:

* *Reduced operational overhead*: You can distribute CA certificates across your cluster as a Day 2 operation.

> 

Reviewer: ++ 
- Combine certificates from multiple sources (ConfigMaps, Secrets, inline, default CAs) into a single bundle
- Automatically update targets when source certificates change

* *Application requirements support*: You can write trust bundles to Secret objects for applications that specifically require secrets instead of ConfigMap objects.

> * *Enhanced security and auditability*: You can use dynamic role-based access control (RBAC) to control secret write permissions and identify all artifacts created by the operator.

Reviewer: > *Enhanced security and auditability*

Sounds a bit too much for trust-manager; I think we can skip all these sub-headings like `*Reduced operational overhead*` or `*Application requirements support*` and write the bullet points directly?

Author: yeah, that works as well.

> * *Enhanced security and auditability*: You can use dynamic role-based access control (RBAC) to control secret write permissions and identify all artifacts created by the operator.

Reviewer: I think it has nothing to do with trust-manager benefits 

```suggestion
* *Enhanced security and auditability*: You can use dynamic role-based access control (RBAC) to control secret write permissions.
```

> * *Automatic CA integration*: The controller automatically reads the trusted CA bundle from the `cert-manager-operator-trusted-ca-bundle` ConfigMap. This ConfigMap is automatically injected by the Cluster Network Operator (CNO) during OLM installation, requiring no manual configurations.

Reviewer: From user's point of view, do they require this much internal details, like which configmap the controller will create or how CNO will inject the CA certs?

Author: Bharath had asked to call this point out. Do you think this isn't required at all or is there an high-level alternative?

Reviewer: WDYT @bharath-b-rh, isn't this bit verbose from the user's point of view?

Author: @bharath-b-rh your thoughts here?

Reviewer 2: The details will be helpful for users to know where the default CA bundle is derived from. For overview if we could keep it one-liner and include the details in API table or `Bundle` CR creation section would be good I think. WDYT?

:_mod-docs-content-type: ASSEMBLY
[id="cert-manager-trust-manager"]
= Distributing certificates by using trust-manager operand
include::_attributes/common-attributes.adoc[]
:context: cert-manager-trust-manager

[role="_abstract"]
The trust-manager operand simplifies the distribution of certificate authority (CA) certificates across {product-title} clusters. As an administrator, you can configure the operand according to the cluster requirements and manage trust bundles efficiently.

:FeatureName: Distributing certificates by using trust manager
include::snippets/technology-preview.adoc[leveloffset=+1]

The trust-manager operand provides the following benefits:

* Distribution of CA certificates across your cluster as a Day 2 operation.

* Consolidation of certificates from multiple sources, such as ConfigMaps, Secrets, inline data, and default CAs, into a single trust bundle.

* Automatic updates to target objects whenever the underlying source certificates change.

> * Creation of trust bundles as Secret objects, directly supporting applications that explicitly require secrets instead of ConfigMap objects.

Reviewer: nit
```suggestion
* Creation of trust bundles as Secret objects, useful for applications that explicitly require secrets instead of ConfigMap objects.
```

> * Dynamic control over secret write permissions by using role-based access control (RBAC).

Reviewer: trust-manager operand itself does not provide this functionality, it's the operator who manages the dynamic RBAC. 

Can we write something like

> cert-manager-operator dynamically controls the secret write permissions provided to the operand based on your configuration.

Author: In that case, is this required to be listed as a benefit? I am guessing there's nothing specific to the operand here.

Reviewer: I am fine with removing this.

> 

Reviewer: Looks like we dropped the defaultCA package benefit in the latest changes. 

https://github.com/openshift/openshift-docs/pull/108188#discussion_r3069129754

Author: I was waiting for Bharath's comment before I put up something.

Reviewer: Let's add a line, otherwise we might forget

include::modules/cert-manager-trust-manager-install.adoc[leveloffset=+1]

[role="_additional-resources"]
.Additional resources
* xref:../../nodes/clusters/nodes-cluster-enabling-features.adoc#nodes-cluster-enabling-features[Enabling features using feature gates]

include::modules/cert-manager-configure-trust-bundle.adoc[leveloffset=+1]

> include::modules/cert-manager-trust-manager-fields.adoc[leveloffset=+1]

Reviewer: Can we change the ordering here? I feel the user should be aware of the field details first, before creating the Bundle CR.

Author: Based on the comment [here](https://github.com/openshift/openshift-docs/pull/108188#discussion_r3069144521), I revisited the standards we usually follow. We are generally placing the fields after all the procedures are done.It is a bit debatable, but having a logical flow of tasks is recommended over disconnected pieces, especially since we are stating something as "Next step". Your question in the above linked comment is something that I expect the users would get too.

:_mod-docs-content-type: ASSEMBLY
[id="cert-manager-trust-manager"]
= Distributing certificates by using trust-manager operand
include::_attributes/common-attributes.adoc[]
:context: cert-manager-trust-manager

[role="_abstract"]
The trust-manager operand simplifies the distribution of certificate authority (CA) certificates across {product-title} clusters. As an administrator, you can configure the operand according to the cluster requirements and manage trust bundles efficiently.

:FeatureName: Distributing certificates by using trust manager
include::snippets/technology-preview.adoc[leveloffset=+1]

The trust-manager operand provides the following benefits:

> * Distribution of CA certificates across your cluster as a Day 2 operation.

Reviewer 3: Day 2 > post-installation operation?

Author: yes

* Consolidation of certificates from multiple sources, such as ConfigMaps, Secrets, inline data, and default CAs, into a single trust bundle.

* Automatic updates to target objects whenever the underlying source certificates change.

> * Creation of trust bundles as Secret objects for applications that explicitly require secrets instead of ConfigMap objects.

Reviewer 3: ```suggestion
* Creation of trust bundles as secret objects for applications that explicitly require secrets instead of ConfigMap objects.
```
